### DISCUSSION TITLE
There is no type resolver defined for interface / union 'XXX' type

### DISCUSSION THREAD

**Discussion by d35h** [graphql-java/graphql-java#2589] [Q&A] [answered]
Hello everyone,

I'm trying to create / validate schema, which comes to me as plain text, and I'm getting errors when trying to makeExecutableSchema, this schema has interfaces and they're a problem, since they have no type resolvers, but I can't create resolvers for them in my code, as my schema isn't static and interfaces might change or new interfaces might appear. 

My code is:
```
val validationRules = ValidationRules.newValidationRules()
            .onValidationErrorStrategy(OnValidationErrorStrategy.RETURN_NULL)
            .build()
        val schemaWiring = ValidationSchemaWiring(validationRules)
        val runtimeWiring = RuntimeWiring
            .newRuntimeWiring()
            .scalar(GraphQLBigDecimal)
            .scalar(DateTime)
            .directiveWiring(schemaWiring)
            .build()
        val typeDefinitionRegistry: TypeDefinitionRegistry = SchemaParser()
            .parse(schema)

        val graphQLSchema = SchemaGenerator().makeExecutableSchema(typeDefinitionRegistry, runtimeWiring)
```
The exception I'm getting is:
```
Internal Server Error: errors=[There is no type resolver defined for interface / union 'Entity' type]
```

My schema is (I can publish only part of it as it's sensitive data):
```
interface Entity {
    id: ID! @unique
}

type Transaction implements Entity
    id: ID!                             @map(source:"TransactionID")
}
```

Can anyone help with this issue, pls?

Thank you in advance,
Dan

**Comment by bbakerman** [comment]
Check out `graphql.schema.idl.RuntimeWiring#MOCKED_WIRING` - it gives you a `RuntimeWiring` that cant be executed BUT will load any SDL - we use this for schema validation in our company.

**Comment by d35h** [comment, marked as answer]
@bbakerman thanks for your reply. I forgot that I have a discussion open, but I managed to solve it already. For those are also having the same problem, here is my code:
```
                val graphQLSchema = SchemaGenerator.createdMockedSchema(schema.readText())
                GraphQL.newGraphQL(graphQLSchema).build()
```
It throws an exception when validation fails
